### ISSUE TITLE
uneven dash distribution (GLMakie)

### ISSUE THREAD

**Issue by JonasIsensee** [MakieOrg/Makie.jl#1614]
Originally reported some time ago in https://discourse.julialang.org/t/makie-uneven-dash-distribution/70100
but I just noticed that this is still the case.

```
using GLMakie
xs = range(-pi, pi, length=1000)
lines(xs, sin.(xs), linestyle=:dot, linewidth=10)
```
![image](https://user-images.githubusercontent.com/33990604/150497750-3d20dbe5-d61f-428e-a80c-4f4af29100ec.png)

This problem appears to be related to the on-screen curvature and slope. Enlarging the plot to e.g. 4000x2000 pixels "removes" the issue. Similarly, zooming in on the problematic portions makes them render differently.
I do not observe this in CairoMakie.

```
julia v1.7.0
GLMakie v0.4.7
```

**Comment by jkrumbiegel** [comment]
My guess would be that it's hard to do this with a gpu drawing routine, as dashes depend on the accumulated line length, which a fragment shader shouldn't know about (I assume). But Cairo does, because it draws sequentially along the line (again, I assume, from my understanding of how CPU vs GPU based drawing works). So GLMakie looks worse if you have many short line segments, because within one quad the logic works, but not across quads. @SimonDanisch correct me if that doesn't make any sense.

**Comment by ffreyer** [comment]
Should be fixed by #2666
